app/assets/javascripts/components/user_profiles/user_profile.jsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import createReactClass from 'create-react-class';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import ContributionStats from './contribution_stats.jsx';
+import CourseDetails from './course_details.jsx';
+import UserUploads from './user_uploads.jsx';
+import { fetchStats } from '../../actions/user_profile_actions.js';
+import Loading from '../common/loading.jsx';
+
+
+const UserProfile = createReactClass({
+  propTypes: {
+    params: PropTypes.object,
+    fetchStats: PropTypes.func.isRequired,
+    stats: PropTypes.object.isRequired,
+    isLoading: PropTypes.bool.isRequired
+  },
+
```

Comment:
I had to add a getInitialState() to set an empty object, to avoid an error from `this.state.statsGraphsData` before the graph data loads.